src/05-derivatives/01-introduction.md

```diff
@@ -0,0 +1,218 @@
+# Goals and scope
+
+Derivatives are outputs of common processing pipelines, capturing data and
+meta-data sufficient for a researcher to understand and (critically) reuse those
+outputs in subsequent processing. Standardizing derivatives is motivated by use
+cases where formalized machine-readable access to processed data enables higher
+level processing.
+
+## Storage of derived datasets
+
+Derivatives can be stored/distributed in two ways:
+
+1.  Under a `derivatives/` subfolder in the root of the source BIDS dataset
+    folder to make a clear distinction between raw data and results of data
+    processing. Each pipeline has a dedicated directory under which it stores
+    all of its outputs. There is no restriction on the directory name; however,
```

Comment:
One question here: We recently had some confusion with [OpenNeuro ds228](https://openneuro.org/datasets/ds000228/versions/1.0.0). The authors generously shared both the raw and derived data, and we noticed in [the derivatives README](https://openneuro.org/datasets/ds000228/versions/1.0.0/file-display/derivatives:README) that, although their derivatives folder was labelled as `fmriprep` the data was in fact processed with `SPM8`. Indeed, the authors very kindly [confirmed this](https://openneuro.org/datasets/ds000228/versions/1.0.0#comment-5c76ce2580ad09406528f831):

> Hello, and thanks for your interest in the data! You are correct that the preprocessed data have been smoothed (and realigned/coregistered and normalized). These data live in a subfolder of the derivatives folder (confusingly currently named fmriprep; preprocessing was done in-house using SPM8 and not fmriprep). 

Would it make sense, then, to put some limits on the pipeline directory names (e.g., such that recognized BIDS Apps are protected values) ?

Reply:
I see the issue, but if we turn the <pipeline_name> into a controlled vocabulary it might cause some friction in adoption. One solution would be to have a set of reserved names for specific pipelines (such as freesurfer, fmriprep etc.). Future bids-specification releases might extend that list with new pipeline names. Unfortunately such semi-controlled vocabulary will be hard to validate automatically.

Reply:
Yes, I think reserved names is a more realistic option !

I see the concern about automatic validation. One idea: could we check that if your pipeline name contains e.g., `fmriprep` then the `PipelineDescription.Name` metadata must be `fmriprep` ?

Reply:
I don't see how this particular example could have been prevented by a controlled/reserved vocabulary... as I understand it,  the user in this case decided to name the folder `fmriprep`, even though there was no reason to call it that—presumably because they thought that was the generic name for preprocessed data. Given that assumption, it would have been a small and natural step to also erroneously set `PipelineDescription.Name` to the same value.

This strikes me as something we can only solve with documentation. Something somewhere needs to make it clear that pipeline folders should be named based on the tool that generated them, and that `fmriprep` isn't a generic name for preprocessed fMRI data (which I'm guessing is what happened here). Given that nobody really likes to read documentation, we probably just have to live with this kind of thing happening once in a while, and addressing it reactively.

Reply:
Absolutely agreed that documentation is an important part of this !

I'm not sure, though, why that's mutually exclusive with enforcing some condition on folder names to be checked in the validator. If the authors indeed thought that `fmriprep` was just a generic name for any pipeline creating preprocessed fMRI data, then it's entirely possible that they would put the same information in the metadata. But it also seems equally likely to me that an author would think that the folder names reflect derivative modalities (i.e., `smriprep`, `dmriprep`) and would note the correct pipeline in the metadata (as these authors did in the README).

So having both documentation _and_ a check in the validator would help to clarify some (I'd argue significant) portion of these cases.

Reply:
Oh, sorry—that wasn't meant as an argument against extra validation checks, but as an argument against having a controlled list of pipeline names. I'm all for checking that the value in `PipelineDescription.Name` is a substring of the folder name.

Reply:
Ah, sorry for the confusion ! Thanks for clarifying.

Then it sounds like we're agreed re enforcing that the `PipelineDescription.Name` field is equal to or a sub-string of the folder name.

As to whether or not those field values should include a controlled list &mdash; I see your point that there's nothing to stop me from (consistently) calling _any_ pipeline an arbitrary name that happens to overlap with an existing pipeline, `fmriprep` being the obvious example.

I think having the [current list of BIDS Apps](http://bids-apps.neuroimaging.io/apps/) as a controlled list seems like a natural thing to suggest, but I'm also struggling to see how we would validate that. So happy to settle with just enforcing the correspondence in folder name and `PipelineDescription.Name` for now ! Or hearing any other thoughts on this point.

Reply:
Done in https://github.com/bids-standard/bids-specification/pull/109/commits/eec7a3e15b403ccbbcff0bc840fdff51078e70a3.